File: csharp/protein-translation/ProteinTranslation.cs
using System.Collections.Generic;
using System.Linq;
public static class ProteinTranslation
{
    private static readonly IReadOnlyDictionary<string, string> proteins = new Dictionary<string, string> {
        {"AUG","Methionine"},
        {"UUU","Phenylalanine"},{"UUC","Phenylalanine"},
        {"UUA","Leucine"},{"UUG","Leucine"},
        {"UCU","Serine"},{"UCC","Serine"},{"UCA","Serine"},{"UCG","Serine"},
        {"UAU","Tyrosine"},{"UAC","Tyrosine"},
        {"UGU","Cysteine"},{"UGC","Cysteine"},
        {"UGG","Tryptophan"},
        {"UAA","STOP"},{"UAG","STOP"},{"UGA","STOP"},
    };
    public static string[] Proteins(string strand) =>
        strand.Where((_, s) => s % 3 == 0).Select((_, s) => new string(strand.Skip(s * 3).Take(3).ToArray())).ToArray()
              .TakeWhile(i => proteins[i] != "STOP").Select(i => proteins[i]).ToArray();

}

File: javascript/protein-translation/protein-translation.js
//
// This is only a SKELETON file for the 'Protein Translation' exercise. It's been provided as a
// convenience to get you started writing code faster.

import { error } from "console";

//
const CODONS = (() => {
  let AUG, UUU, UUC, UUA, UUG, UCU, UCC, UCA, UCG, UAU, UAC, UGU, UGC, UGG, UAA, UAG, UGA
  AUG = 'Methionine'
  UUU = UUC = 'Phenylalanine'
  UUA = UUG = 'Leucine'
  UCU = UCC = UCA = UCG = 'Serine'
  UAU = UAC = 'Tyrosine'
  UGU = UGC = 'Cysteine'
  UGG = 'Tryptophan'
  UAA = UAG = UGA = 'STOP'
  return { AUG, UUU, UUC, UUA, UUG, UCU, UCC, UCA, UCG, UAU, UAC, UGU, UGC, UGG, UAA, UAG, UGA }
})();

export const translate = (rna = "") =>
  (rna.match(/.{1,3}/g) || []).reduce((arr, codon) => {
    if (!CODONS[codon])
      throw new Error('Invalid codon');
    return arr.includes('STOP') ? arr : [...arr, CODONS[codon]];
  }, [])
    .filter(codon => codon !== 'STOP');